designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputDropDown.kt

```diff
@@ -33,22 +52,138 @@ import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
  * item is focused
  * @param modifier allows a modifier to be passed externally
  * @param onResetButtonClicked callback to when reset button is clicked
- * @param onArrowDropDownButtonClicked callback to when arrow drop down button is clicked
  */
+@OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun InputDropDown(
     title: String,
     state: InputShellState,
-    selectedItem: String? = null,
+    dropdownItems: List<DropdownItem>,
+    selectedItem: DropdownItem? = null,
     supportingTextData: List<SupportingTextData>? = null,
     legendData: LegendData? = null,
     isRequiredField: Boolean = false,
     modifier: Modifier = Modifier,
     onFocusChanged: ((Boolean) -> Unit)? = null,
     onResetButtonClicked: () -> Unit,
-    onArrowDropDownButtonClicked: () -> Unit,
+    onItemSelected: (DropdownItem) -> Unit,
 ) {
     val focusRequester = remember { FocusRequester() }
+    var showDropdown by remember { mutableStateOf(false) }
+
+    val inputField: @Composable (modifier: Modifier) -> Unit = { inputModifier ->
+        InputField(
+            modifier = inputModifier,
+            focusRequester = focusRequester,
+            title = title,
+            state = state,
+            isRequiredField = isRequiredField,
+            expanded = showDropdown,
+            onFocusChanged = onFocusChanged,
+            supportingTextData = supportingTextData,
+            legendData = legendData,
+            selectedItem = selectedItem,
+            onResetButtonClicked = onResetButtonClicked,
+            onDropdownIconClick = {
+                showDropdown = !showDropdown
+            },
+        )
+    }
+
+    if (dropdownItems.size > MAX_DROPDOWN_ITEMS) {
+        Box {
+            inputField(modifier)
+
+            if (showDropdown) {
+                BottomSheetShell(
```

Comment:
Not sure, but should we need the searchbar to meet the requirements?
![image](https://github.com/dhis2/dhis2-mobile-ui/assets/3330344/737a9cbd-c1eb-40f1-8368-9e9316c80e8f)

Reply:
Good point, let me confirm that with Marcos and make that change if required.

Reply:
Made the change